src/main/java/com/easypost/model/AddressCollection.java

```diff
@@ -1,10 +1,25 @@
 package com.easypost.model;
 
 import java.util.List;
+import java.util.Map;
+
 import lombok.Getter;
 
 @Getter
-public final class AddressCollection extends EasyPostResource {
+public final class AddressCollection extends PaginatedCollection<Address> {
     private List<Address> addresses;
-    private Boolean hasMore;
+
+    @Override
+    protected Map<String, Object> buildNextPageParameters(List<Address> addresses, Integer pageSize) {
+        String lastId = addresses.get(addresses.size() - 1).getId();
+
+        Map<String, Object> parameters = new java.util.HashMap<>();
```

Comment:
We certainly can, what's the rationale?
The `.all()` function that these parameters are getting passed into accepts the base `Map` type rather than a specific `HashMap`

Reply:
Just personal nit, I prefer to import it before using it, but this works